### PR TITLE
Feature/longitudinal

### PR DESCRIPTION
**Pull request by SucyLi**
anatomical longitudinal workflow

### CODE REVIEW

File: CPAC/registration/registration.py [hunk 1]
```diff
@@ -927,3 +926,332 @@ def create_wf_calculate_ants_warp(name='create_wf_calculate_ants_warp', num_thre
...
+                'mean_functional_in_anat': (func_to_anat, 'outputspec.anat_func_nobbreg'),
+                'functional_to_anat_linear_xfm': (func_to_anat, 'outputspec.func_to_anat_linear_xfm_nobbreg')
+            })
+
+    strat_list += new_strat_list
+
+    return workflow, strat_list, diff_complete
+
+
+def connect_func_to_anat_bbreg(workflow, strat_list, c, diff_complete):
```
Comment (sgiavasis) [inline CPAC/registration/registration.py:1022]: 👍

File: CPAC/utils/datasource.py [hunk 2]
```diff
@@ -507,16 +508,22 @@ def resolve_resolution(resolution, template, template_name, tag = None):
...
     if "{" in template and tag is not None:
             tagname = "${" + tag + "}"
     try:
         if tagname is not None:
             local_path = check_for_s3(template.replace(tagname, str(resolution)))     
-    except IOError:
+    except (IOError, OSError):
         local_path = None
 
+    ## TODO debug - it works in ipython but doesn't work in nipype wf
```
Comment (sgiavasis) [inline CPAC/utils/datasource.py:521]: Anything necessary?

File: CPAC/anat_preproc/anat_preproc.py [hunk 3]
```diff
@@ -130,13 +151,40 @@ def create_anat_preproc(method='afni', already_skullstripped=False, config=None,
     else:
         preproc.connect(anat_deoblique, 'out_file', anat_reorient, 'in_file')
 
+    # TODO reassess average center of mass implementation, esp for skull alignment
+    '''
+    if 'anat' in config.run_longitudinal:
+        anat_align_cmass = pe.Node(interface=afni.CenterMass(), name='cmass')
```
Comment (shnizzedy) [inline CPAC/anat_preproc/anat_preproc.py:157]: I got 

```Python
200711-19:44:42,832 nipype.workflow INFO:
   [Node] Running "template_skull_for_anat_center_of_mass" ("nipype.interfaces.afni.utils.CenterMass"), a CommandLine Interface with command:
3dCM /media/ebs/runs/2020-07-11_feature-longitudinal/working/anat_longitudinal_template_0027234/template_skull_for_anat_center_of_mass/MNI152_T1_2mm.nii.gz > template_center_of_mass.txt
200711-19:44:42,834 nipype.workflow WARNING:
   [Node] Error on "anat_longitudinal_template_0027234.template_skull_for_anat_center_of_mass" (/media/ebs/runs/2020-07-11_feature-longitudinal/working/anat_longitudinal_template_0027234/template_skull_for_anat_center_of_mass)
200711-19:44:42,835 nipype.workflow ERROR:
   Node template_skull_for_anat_center_of_mass failed to run on host 8a09b9689bb3.
200711-19:44:42,835 nipype.workflow ERROR:
   Saving crash info to /media/ebs/runs/2020-07-11_feature-longitudinal/crash/crash-20200711-194442-UID1000-template_skull_for_anat_center_of_mass-1cce622e-38d6-4917-8237-d53d7a5f3461.pklz
Traceback (most recent call last):
File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/linear.py", line 44, in run
  node.run(updatehash=updatehash)
File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 471, in run
  result = self._run_interface(execute=True)
File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 555, in _run_interface
  return self._run_command(execute)
File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 635, in _run_command
  result = self._interface.run(cwd=outdir)
File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 521, in run
  runtime = self._run_interface(runtime)
File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/afni/base.py", line 110, in _run_interface
  return super(AFNICommandBase, self)._run_interface(runtime)
File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 1024, in _run_interface
  runtime.hostname))
OSError: No command "3dCM" found on host 8a09b9689bb3. Please check that the corresponding package is installed.
```

https://github.com/FCP-INDI/C-PAC/pull/1320 fixed the issue for me

File: CPAC/anat_preproc/anat_preproc.py [hunk 4]
```diff
@@ -491,3 +556,232 @@ def create_anat_preproc(method='afni', already_skullstripped=False, config=None,
             preproc.connect(refined_brain, 'out_file', outputnode, 'brain')
 
     return preproc
+
+
+def connect_anat_segmentation(workflow, strat_list, c, strat_name=None):
```
[the remark this answers is not in the document]
Comment (SucyLi) [reply CPAC/anat_preproc/anat_preproc.py]: Sure! moved it to seg_preproc.py

File: CPAC/anat_preproc/anat_preproc.py [hunk 5]
```diff
@@ -130,13 +151,40 @@ def create_anat_preproc(method='afni', already_skullstripped=False, config=None,
...
+    else:
+    '''
     preproc.connect(anat_reorient, 'out_file', outputnode, 'reorient')
 
     if already_skullstripped:
 
         anat_skullstrip = pe.Node(interface=util.IdentityInterface(fields=['out_file']),
                                     name='anat_skullstrip')
 
+        # if 'anat' in config.run_longitudinal:
```
[the remark this answers is not in the document]
Comment (SucyLi) [reply CPAC/anat_preproc/anat_preproc.py]: removed

**Comment by sgiavasis** [comment]
Great!

**Pull request merged**